File: service/src/main/java/top/emilejones/hhu/service/strategy/RecallStrategy.java
package top.emilejones.hhu.service.strategy;

import kotlin.Pair;
import top.emilejones.hhu.entity.FileNode;
import top.emilejones.hhu.entity.TextNode;

import java.util.List;

public interface RecallStrategy {
    List<Pair<FileNode, TextNode>> exec(List<Pair<FileNode, TextNode>> rawData);
}

File: infrastructure/textsplitter/src/main/kotlin/top/emilejones/hhu/textsplitter/service/strategy/RecallStrategy.kt
package top.emilejones.hhu.textsplitter.service.strategy

import top.emilejones.hhu.textsplitter.domain.po.Neo4jFileNode
import top.emilejones.hhu.textsplitter.domain.po.Neo4jTextNode

interface RecallStrategy {
    fun exec(rawData: List<Pair<Neo4jFileNode, Neo4jTextNode>>): List<Pair<Neo4jFileNode, Neo4jTextNode>>
}
